File: src/main/java/com/adjorno/billib/rest/db/Artist.java
package com.adjorno.billib.rest.db;

import javax.persistence.*;

@Entity
@Table(name = "ARTIST")
public class Artist {
    @Id
    @Column(name = "_id")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long mId;

    @Column(name = "NAME")
    private String mName;

    public Artist() {
    }

    public Artist(Long id, String name) {
        mId = id;
        mName = name;
    }

    public Long getId() {
        return mId;
    }

    public void setId(Long id) {
        mId = id;
    }

    public String getName() {
        return mName;
    }

    public void setName(String name) {
        mName = name;
    }

    @Override
    public String toString() {
        return mName;
    }

    @Override
    public boolean equals(Object theo) {
        if (this == theo) {
            return true;
        }
        if (theo == null || getClass() != theo.getClass()) {
            return false;
        }

        Artist thetheArtist = (Artist) theo;

        return mId.equals(thetheArtist.mId);
    }

    @Override
    public int hashCode() {
        return mId.hashCode();
    }

}

File: src/main/java/com/adjorno/billib/rest/db/Artist.kt
package com.adjorno.billib.rest.db

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "ARTIST")
data class Artist(
    @Id
    @Column(name = "_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(name = "NAME")
    var name: String? = null,

    // Generated column for normalized search (read-only)
    @Column(name = "NAME_NORMALIZED", insertable = false, updatable = false)
    var nameNormalized: String? = null
) {
    fun generateDuplicateTitle(title: String): String {
        return "$name - $title"
    }

    override fun toString(): String = name.toString()
}
